### PR TITLE
fixes #8058 tests(nimbus): Update enrollment telemetry integration test

### LINKED ISSUE #8058: Update our older telemetry integration test to the newer methods
Now that the pref flip test is done and stable, lets add those improvements to the enroll/unenroll test.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3004)

### PR DESCRIPTION
**Pull request by jrbenny35**
Because:
* With the new pref-flip integration test, we found ways to improve on a few things

This commit:
* Adds those changes to the old test
* Updates the logic for the pref flip test helper a bit

### CODE REVIEW

**Comment by jrbenny35** [comment]
The circleci Enrollment test job went from 15m to 4! 🔥

File: app/tests/integration/nimbus/test_desktop_client_integrations.py [hunk 1]
```diff
@@ -124,31 +123,12 @@ def test_check_telemetry_enrollment_unenrollment(
...
-                            break
-            else:
-                continue
-        # If there are no pings we have to wait
-        else:
-            trigger_experiment_loader()
-            time.sleep(15)
-    else:
-        if control is not False:
+    while control is not True:
```
Comment (yashika-khurana) [inline app/tests/integration/nimbus/test_desktop_client_integrations.py:126]: why not just write  `while control`?

File: app/tests/integration/nimbus/test_desktop_client_integrations.py [hunk 2]
```diff
@@ -172,31 +152,12 @@ def test_check_telemetry_enrollment_unenrollment(
...
-            else:
-                continue
-        # If there are no pings we have to wait
-        else:
-            trigger_experiment_loader()
-            time.sleep(15)
-    else:
-        if control is not False:
-            assert False, "Experiment unenrollment was never seen in Ping Data"
+    while control is not True:
```
Comment (yashika-khurana) [inline app/tests/integration/nimbus/test_desktop_client_integrations.py:155]: same as above ⬆️

**Review by yashika-khurana — CHANGES_REQUESTED** [review]
Overall looks good to me, thank you @jrbenny35 just a minor suggestion

File: app/tests/integration/nimbus/test_desktop_client_integrations.py (hunk 1 above)
Comment (jrbenny35) [reply app/tests/integration/nimbus/test_desktop_client_integrations.py:126]: Oh this is wrong. Control should be false. Fixing

Comment (yashika-khurana) [reply app/tests/integration/nimbus/test_desktop_client_integrations.py:126]: then `While not control`

**Review by jaredlockhart — APPROVED** [review]
Hot dog!!!  🎉 🎉 🎉 🎉 🎉

**Pull request merged**